### PR TITLE
UOL - BidAdapter release candidate

### PR DESCRIPTION
**Pull request by lotani-uolinc**
## Type of change
- [x] New bidder adapter 

## Description of change
New bidAdapter proposal from UOL - Universo Online.

- test parameters for validating bids
```
{
  bidder: 'uol',
  params: {
    placementId: 1231244,
    test: true,
    cpmFactor: 2
  }
}
``

- contact email of the adapter’s maintainer
   l-prebid@uolinc.com

- [x] official adapter submission

### CODE REVIEW

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging fc26641bfe46267fdb198ec1e4622b47fb641e66 into ed28094054145e132d889d36ad79d51a00210c63 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-7ca060a1a7ea9de31c1665cd7c35698beab550cf)

**new alerts:**

* 1 for Superfluous trailing arguments

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jsnellbaker** [comment]
@lotani-uolinc Can you please look into the LGTM error when you get the chance?

File: modules/uolBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+import * as utils from 'src/utils';
+// import {config} from 'src/config';
```
Comment (jsnellbaker) [inline modules/uolBidAdapter.js]: this comment can be removed.

**Comment by lotani-uolinc** [comment]
@jsnellbaker, sure thing! Thanks for the feedback.

File: modules/uolBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'uol';
+const ENDPOINT_URL = 'https://prebid.adilligo.com/v1/prebid.json';
+const UOL_LOG_HEADER = 'UOL Bidder Error: '
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['banner'], // not required since it is default
```
Comment (jsnellbaker) [inline modules/uolBidAdapter.js]: Could import/use the defined mediaTypes from `src/mediaTypes`.  Eg `import { BANNER } from 'src/mediaTypes';`  This way you can ensure the values are read correctly.

File: modules/uolBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  /**
+     * Register the user sync pixels which should be dropped after the auction.
+     *
+     * @param {SyncOptions} syncOptions Which user syncs are allowed?
+     * @param {ServerResponse[]} serverResponses List of server's responses.
+     * @return {UserSync[]} The user syncs which should be dropped.
+     */
+  getUserSyncs: function(syncOptions, serverResponses) {
+    const syncs = [];
+    if (syncOptions.iframeEnabled && serverResponses.body.trackingPixel) {
```
Comment (jsnellbaker) [inline modules/uolBidAdapter.js]: `serverResponses` is an array of objects, not a single object.  This latter half of the code would result in undefined errors.  Could you update this logic?

File: modules/uolBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    return syncs;
+  },
+
+  /**
+    * Register bidder specific code, which will execute if bidder timed out after an auction
+    * @param {data} Containing timeout specific data
+    */
+  onTimeout: function(data) {
+    // Bidder specifc code
+    return true;
```
Comment (jsnellbaker) [inline modules/uolBidAdapter.js]: Is there any more you would wish to do in the case of a timeout?

**Comment by jsnellbaker** [comment]
@lotani-uolinc I saw your response in github before in regards to GDPR support, but I'm not seeing it anymore (not sure what happened).  

If you're still not currently planning on  supporting GDPR - could you please make a change to your docs PR?

The field in the docs PR is an internal variable for the prebid site that will list that adapter in a certain table of GDPR compliant adapters.  Until your adapter supports GDPR, can you either remove the variable or set it to `false`?

**Comment by lotani-uolinc** [comment]
@jsnellbaker, no problem. We have set it to false and will release it soon. Thanks in advance.

**Comment by lotani-uolinc** [comment]
@jsnellbaker, we have addressed the changes requested. Could you please evaluate?
Thanks!

File: modules/uolBidAdapter.js (hunk 4 above)
Comment (lotani-uolinc) [reply modules/uolBidAdapter.js]: For the time being we have no plans hence, Im removing the method call.

**Comment by jsnellbaker** [comment]
@lotani-uolinc Thanks for making the updates.  They look fine to me.  In addition the test param information seems to be working successfully as well.

Something I noticed while testing with browserstack in preparation for the merge were a series of errors from the unit tests.  There seemed to be two sets of errors occurring in different browsers (safari mainly) that I've copied down below for reference.  Can you take a look and see if something can be done to address these errors?

```
UOL Bid Adapter
    buildRequests
      buildRequest geolocation param
        [31m✗ [39m[31mshould contain user coordinates if (i) DNT is off; (ii) browser supports implementation; (iii) localStorage contains geolocation history[39m
	undefined is not an object (evaluating 'navigator.permissions.query')
	grantTriangulation@webpack:///test/spec/modules/uolBidAdapter_spec.js:96:46 <- test/test_index.js:129093:46
	webpack:///test/spec/modules/uolBidAdapter_spec.js:177:8 <- test/test_index.js:129172:27

        [31m✗ [39m[31mshould not contain user coordinates if localStorage is empty[39m
	undefined is not an object (evaluating 'navigator.permissions.query')
	denyTriangulation@webpack:///test/spec/modules/uolBidAdapter_spec.js:104:46 <- test/test_index.js:129101:46
	webpack:///test/spec/modules/uolBidAdapter_spec.js:186:8 <- test/test_index.js:129181:26

        [31m✗ [39m[31mshould not contain user coordinates if browser doesnt support permission query[39m
	undefined is not an object (evaluating 'navigator.permissions.query')
	removeQuerySupport@webpack:///test/spec/modules/uolBidAdapter_spec.js:112:46 <- test/test_index.js:129109:46
	webpack:///test/spec/modules/uolBidAdapter_spec.js:195:8 <- test/test_index.js:129190:27
```

```
UOL Bid Adapter
    buildRequests
      buildRequest geolocation param
        [31m✗ [39m[31mshould contain user coordinates if (i) DNT is off; (ii) browser supports implementation; (iii) localStorage contains geolocation history[39m
	TypeError: Unable to get property 'query' of undefined or null reference
	   at grantTriangulation (webpack:///test/spec/modules/uolBidAdapter_spec.js:96:6 <- test/test_index.js:129093:7)
	   at Anonymous function (webpack:///test/spec/modules/uolBidAdapter_spec.js:177:8 <- test/test_index.js:129172:9)

        [31m✗ [39m[31mshould not contain user coordinates if localStorage is empty[39m
	TypeError: Unable to get property 'query' of undefined or null reference
	   at denyTriangulation (webpack:///test/spec/modules/uolBidAdapter_spec.js:104:6 <- test/test_index.js:129101:7)
	   at Anonymous function (webpack:///test/spec/modules/uolBidAdapter_spec.js:186:8 <- test/test_index.js:129181:9)

        [31m✗ [39m[31mshould not contain user coordinates if browser doesnt support permission query[39m
	TypeError: Unable to get property 'query' of undefined or null reference
	   at removeQuerySupport (webpack:///test/spec/modules/uolBidAdapter_spec.js:112:6 <- test/test_index.js:129109:7)
	   at Anonymous function (webpack:///test/spec/modules/uolBidAdapter_spec.js:195:8 <- test/test_index.js:129190:9)
```

**Comment by lotani-uolinc** [comment]
@jsnellbaker, thanks for the review. We are now looking into alternatives for the safari issue.

**Comment by lotani-uolinc** [comment]
@jsnellbaker, as a fix, we disabled geolocation tests for engines that do not implement permission and geolocation APIs.
Feel free to reach if any changes are required.

**Comment by jsnellbaker** [comment]
@lotani-uolinc That's fine; it looks like everything is settled.

**Comment by lotani-uolinc** [comment]
@jsnellbaker, nice! Thanks for your time.

**Pull request merged**